src/snapshot-io/index.ts

```diff
@@ -0,0 +1,244 @@
+      },
+      'proposals'
+    );
+    return response.map((response: any) => ({
+      id: response.id,
+      type: response.type,
+      author: response.author,
+      title: response.title,
+      body: response.body,
+      ipfs: response.ipfs,
+      choices: response.choices,
+      created: new Date(response.created * 1000),
+      start: new Date(response.start * 1000),
+      end: new Date(response.end * 1000),
+      state: response.state,
+      network: response.network,
+      snapshot: response.snapshot,
+      scores: response.scores,
+      votes: response.votes,
+    }));
+  }
+
+  async getProposal(proposalId: string): Promise<SnapshotProposal> {
+    const response = await this.graphQLQuery(
+      PROPOSAL_QUERY,
+      {
+        id: proposalId,
+      },
+      'proposal'
+    );
+
+    return {
+      id: response.id,
+      type: response.type,
+      author: response.author,
+      title: response.title,
+      body: response.body,
+      ipfs: response.ipfs,
+      choices: response.choices,
+      created: new Date(response.start * 1000),
+      start: new Date(response.start * 1000),
+      end: new Date(response.end * 1000),
+      state: response.state,
+      network: response.network,
+      snapshot: response.snapshot,
+      scores: response.scores,
+      votes: response.votes,
+    };
+  }
+
+  async listVotes(
+    proposalId: string,
+    from = 0,
+    count = 30000,
+    voter = ''
+  ): Promise<SnapshotVote[]> {
+    const response = await this.graphQLQuery(
+      VOTES_QUERY,
+      {
+        id: proposalId,
+        orderBy: 'vp',
+        orderDirection: 'desc',
+        first: count,
+        voter,
+        skip: from,
+      },
+      'votes'
+    );
+
+    return response.map((vote: any) => ({
+      voter: vote.voter,
+      choice: vote.choice,
+      power: vote.vp,
+    }));
+  }
+
+  async getERCBalanceOf(
+    spaceId: string,
+    network: string,
+    snapshot: number,
+    token: string,
+    decimals: number,
+    symbol: string,
+    voter: string
+  ): Promise<number> {
+    const strategies = this.generateStrategies(token, decimals, symbol);
+
+    let scores: any = await Client.utils.getScores(
+      spaceId,
+      strategies,
+      network,
+      [voter],
+      snapshot
+    );
+    scores = scores.map((score: any) =>
+      Object.values(score).reduce((a, b: any) => a + b, 0)
+    );
+    return scores.reduce((a: number, b: number) => a + b, 0);
+  }
+
+  async createProposal(
+    signer: ethers.Wallet,
+    spaceId: string,
+    title: string,
+    body: string | undefined,
+    choices: string[],
+    duration: number,
+    snapshot: number,
+    network: string,
+    sender: string,
+    recipient: string,
+    token: string,
+    decimals: number,
+    symbol: string,
+    amount: string
+  ): Promise<{
+    id: string;
+    ipfs: string;
+  }> {
+    const startDateTime = new Date();
+    const response: any = await this._clientEIP712.proposal(
```

Comment:
No any types

Reply:
Actually maybe we can let this slide for now since we aren't using snapshot long term.

Since the return of this is `unknown` and we want to interact with it (by getting `.ipfsHash` and `.id`) the proper thing to do is to cast it into what we want, and interrogate the type;

```
interface SnapshotProposalResponse {
  id?: string;
  ipfsHash?: string;
}


/// Usage

const response: SnapshotProposalResponse = await this._clientEIP712.proposal(...) as SnapshotProposalResponse ;

if (!response.id || !response.ipfsHash) {
  // throw an error here
}

return { id: response.id, ipfs: response.ipfsHash }
```

The important thing here is to **not** use `SnapshotProposalResponse` as the return type of our `createProposal` method cause that would be bad code smell.

If `createProposal(...)` returned `CreatedProposal`
You could do `type CreatedProposal = SnapshotProposalResponse` if they just happen to be the same though.